### DISCUSSION TITLE
Fixing usability of nested lattice types in the `#[derive(Lattice)]` macro

### DISCUSSION THREAD

**Discussion by MingweiSamuel** [hydro-project/hydro#1388] [Ideas]
Rohit ran into this - want to use the `#[derive(Lattice)]` macro, but then you can't use the inner fields for a `LatticeBimorphism`

Possible fixes:
1. Allow mapping of `state` references - something we have thought about for a long time. If we could do something like `state() -> map(|x| &x.system)` to map the _state reference_ (not the items) and use that reference in the lattice bimorphism, that work work.
2. Allow some sort of `#[derive(LatticeBimorphism)]` macro in the same was as deriving `Lattice`. But kinda clunky, have to align fields.

**Comment by rohitkulshreshtha** [comment]
Next steps: Design needed. Move to a feature branch until we have a design ready.
